app/drivers/zephyr/kscan_gpio_demux.c

```diff
@@ -0,0 +1,270 @@
+    static int kscan_gpio_disable_##n(struct device *dev) {                                        \
+        LOG_DBG("KSCAN API disable");                                                              \
+        struct kscan_gpio_data_##n *data = dev->driver_data;                                       \
+        k_timer_stop(&data->poll_timer);                                                           \
+        return 0;                                                                                  \
+    };                                                                                             \
+                                                                                                   \
+    /* GPIO init function*/                                                                        \
+    static int kscan_gpio_init_##n(struct device *dev) {                                           \
+        LOG_DBG("KSCAN GPIO init");                                                                \
+        struct kscan_gpio_data_##n *data = dev->driver_data;                                       \
+        int err;                                                                                   \
+        /* configure input devices*/                                                               \
+        struct device **input_devices = kscan_gpio_input_devices_##n(dev);                         \
+        for (int i = 0; i < INST_MATRIX_INPUTS(n); i++) {                                          \
+            const struct kscan_gpio_item_config *in_cfg = &kscan_gpio_input_configs_##n(dev)[i];   \
+            input_devices[i] = device_get_binding(in_cfg->label);                                  \
+            if (!input_devices[i]) {                                                               \
+                LOG_ERR("Unable to find input GPIO device");                                       \
+                return -EINVAL;                                                                    \
+            }                                                                                      \
+            err = gpio_pin_configure(input_devices[i], in_cfg->pin, GPIO_INPUT | in_cfg->flags);   \
+            if (err) {                                                                             \
+                LOG_ERR("Unable to configure pin %d on %s for input", in_cfg->pin, in_cfg->label); \
+                return err;                                                                        \
+            } else {                                                                               \
+                LOG_DBG("Configured pin %d on %s for input", in_cfg->pin, in_cfg->label);          \
+            }                                                                                      \
+            irq_callbacks_##n[i].work = &data->work;                                               \
```

Comment:
If we're using timers and polling because of the IO expander, it seems like we can ditch all of the interrupt setup here, and remove the callback handlers being stored, etc.